Sources/Testing/CMakeLists.txt

```diff
@@ -94,6 +94,11 @@ add_library(Testing
   Traits/Trait.swift)
 target_link_libraries(Testing PRIVATE
   _TestingInternals)
+if(NOT BUILD_SHARED_LIBS)
+  # Merge private dependencies into single static objects archive
+  set_property(TARGET Testing PROPERTY STATIC_LIBRARY_OPTIONS
+    $<TARGET_OBJECTS:_TestingInternals>)
```

Comment:
Is there a reasons not to use `target_link_options(Testing $<TARGET_OBJECTS:_TestingInternals>)`?

Reply:
> Note This command cannot be used to add options for static library targets, since they do not use a linker. To add archiver or MSVC librarian flags, see the [STATIC_LIBRARY_OPTIONS](https://cmake.org/cmake/help/latest/prop_tgt/STATIC_LIBRARY_OPTIONS.html#prop_tgt:STATIC_LIBRARY_OPTIONS) target property.

https://cmake.org/cmake/help/latest/command/target_link_options.html

Yes, it's intentional. Options specified to `target_link_options` are passed only when linking but not archiving, so we need to use `STATIC_LIBRARY_OPTIONS` here.